apps/emqx_rule_engine/src/emqx_rule_engine_jwt_sup.erl

```diff
@@ -0,0 +1,90 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2022 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_rule_engine_jwt_sup).
+
+-behaviour(supervisor).
+
+-export([ start_link/0
+        , start_worker/2
+        , stop_worker/1
+        ]).
+
+-export([init/1]).
+
+-include_lib("emqx_rule_engine/include/rule_actions.hrl").
+
+-type worker_id() :: term().
+
+start_link() ->
+    supervisor:start_link({local, ?MODULE}, ?MODULE, []).
+
+init([]) ->
+    ensure_jwt_table(),
+    SupFlags = #{ strategy => one_for_one
+                , intensity => 10
+                , period => 5
+                , auto_shutdown => never
+                },
+    ChildSpecs = [],
+    {ok, {SupFlags, ChildSpecs}}.
+
+%% @doc Starts a new JWT worker.  The worker will send the caller a
+%% message when it creates and stores its first JWT, or if it fails to
+%% do so, using a generated reference.
```

Comment:
Is this for testing?
can we simplify this to two steps?
1. {ok, Pid} = start_worker(....),
2. gen_server:cast(Pid, {send_me_token, self()}).

Reply:
Not for testing: this is being used in the rule action to synchronize the startup, so we don't finish successfully the resource startup until we have a JWT in ETS.

Reply:
:heavy_check_mark: